ASoC: SOF: Correct sps->stream and cstream nullity management.

ujfalusi opened the pull request:
Hi,

Alternative PR to #5214

The Nullity of sps->cstream needs to be checked in sof_ipc_msg_data() and not assume that it is not NULL.
The sps->stream must be cleared to NULL on close since this is used as a check to see if we have active PCM stream.

ujfalusi committed:
ASoC: SOF: stream-ipc: Check for cstream nullity in sof_ipc_msg_data()

The nullity of sps->cstream should be checked similarly as it is done in
sof_set_stream_data_offset() function.
Assuming that it is not NULL if sps->stream is NULL is incorrect and can
lead to NULL pointer dereference.

Fixes: ef8ba9f79953 ("ASoC: SOF: Add support for compress API for stream data/offset")
Cc: stable@vger.kernel.org
Reported-by: Curtis Malainey <cujomalainey@chromium.org>
Closes: https://github.com/thesofproject/linux/pull/5214
Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi committed:
ASoC: SOF: pcm: Clear the susbstream pointer to NULL on close

The spcm->stream[substream->stream].substream is set during open and was
left untouched. After the first PCM stream it will never be NULL and we
have code which checks for substream NULLity as indication if the stream is
active or not.
For the compressed cstream pointer the same has been done, this change will
correct the handling of PCM streams.

Fixes: ef8ba9f79953 ("ASoC: SOF: Add support for compress API for stream data/offset")
Cc: stable@vger.kernel.org
Reported-by: Curtis Malainey <cujomalainey@chromium.org>
Closes: https://github.com/thesofproject/linux/pull/5214
Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

lgirdwood reviewed (commented):
@cujomalainey pls review - IIUC this fixes and aligns flows with compressed streams.

cujomalainey reviewed (commented):
add my reported-by tags then LGTM

ujfalusi commented:
Changes since v1:
- Added reported-by tag
- Added FIxes tag and Cc stable
- Added link tot he original PR as reference (needed for the reported-by tag)
